lib/index.ts

```diff
@@ -57,19 +67,28 @@ const getImagesToRemove = function (
 	// This should avoid trying to remove images with children.
 	const result = [];
 	let size = 0;
-	while (size < reclaimSpace) {
-		const leafs = getUnusedTreeLeafs(tree).sort((a, b) => {
+	const leafs = getUnusedTreeLeafs(tree);
+	const resort = () => {
+		leafs.sort((a, b) => {
 			// mtime asc, size desc
 			return mtimeSort(a, b) || -sizeSort(a, b);
 		});
+	};
+	resort();
+	while (size < reclaimSpace) {
 		if (leafs.length === 0) {
 			break;
 		}
-		const leaf = leafs[0];
+		const leaf = leafs.shift()!;
```

Comment:
Just confirming that it's safe to shift the zeroth element off of the array? From what I can tell it should be, and would be better to do so to reduce the size of the array for any subsequent processing.

Unrelated to this PR, but I noticed what we can mark the current `leaf` as `removed = true` below even if its equal to `tree` but we won't add it to `result` if its equal to `tree`. This doesn't make sense to me.

Reply:
We do an `if (leafs.length === 0) {` check right above so we know the array can't be empty which means it's safe from that regards (won't be undefined). As for the removal it is now required so we don't keep considering the same element over and over - previously we rebuilt the entire array which had the effect of removing the removed node and reconsidering it's parent, just at great cost for the rest of the tree that was unmodified.

We don't consider the root node for removal because it is a dummy node that doesn't actually exist and just serves to attach all the 1st layer images to, I guess you could count it as analogous to eg `/var/lib/docker`, either way it's not something that can actually be removed